python/lsst/ts/standardscripts/auxtel/mock/attcs_mock.py

```diff
@@ -0,0 +1,296 @@
+            self.task_list.append(asyncio.create_task(self.fake_open_shutter()))
+        elif not id_data.open and self.dome_shutter_pos == 1.:
+            self.task_list.append(asyncio.create_task(self.fake_close_shutter()))
+        else:
+            raise RuntimeError(f"Cannot execute operation: {id_data.open} with dome "
+                               f"at {self.dome_shutter_pos}")
+
+    async def close_shutter_callback(self, id_data):
+        if self.dome_shutter_pos == 1.:
+            self.task_list.append(asyncio.create_task(self.fake_close_shutter()))
+        else:
+            raise RuntimeError(f"Cannot close dome with dome "
+                               f"at {self.dome_shutter_pos}")
+
+    async def fake_open_shutter(self):
+        self.atdome.evt_shutterInPosition.set_put(inPosition=False,
+                                                  force_output=True)
+        self.atdome.evt_mainDoorState.set_put(state=ATDome.ShutterDoorState.OPENING)
+        for self.dome_shutter_pos in np.linspace(0., 1., 10):
+            self.atdome.tel_position.set_put(mainDoorOpeningPercentage=self.dome_shutter_pos)
+            await asyncio.sleep(self.slew_time/10.)
+        self.atdome.evt_shutterInPosition.set_put(inPosition=True,
+                                                  force_output=True)
+        self.atdome.evt_mainDoorState.set_put(state=ATDome.ShutterDoorState.OPENED)
+
+    async def fake_close_shutter(self):
+        self.atdome.evt_shutterInPosition.set_put(inPosition=False,
+                                                  force_output=True)
+        self.atdome.evt_mainDoorState.set_put(state=ATDome.ShutterDoorState.CLOSING)
+        for self.dome_shutter_pos in np.linspace(1., 0., 10):
+            self.atdome.tel_position.set_put(mainDoorOpeningPercentage=self.dome_shutter_pos)
+            await asyncio.sleep(self.slew_time/10.)
+        self.atdome.evt_shutterInPosition.set_put(inPosition=True,
+                                                  force_output=True)
+        self.atdome.evt_mainDoorState.set_put(state=ATDome.ShutterDoorState.CLOSED)
+
+    def get_start_callback(self, comp):
+
+        def callback(id_data):
+
+            ss = getattr(self, comp).evt_summaryState.data.summaryState
+
+            if ss != salobj.State.STANDBY:
+                raise RuntimeError(f"Current state is {salobj.State(ss.summaryState)}.")
+
+            getattr(self, comp).evt_summaryState.set_put(summaryState=salobj.State.DISABLED)
+
+            self.settings_to_apply[comp] = id_data.settingsToApply
+
+        return callback
+
+    def get_enable_callback(self, comp):
+
+        def callback(id_data):
+            ss = getattr(self, comp).evt_summaryState.data.summaryState
+
+            if ss != salobj.State.DISABLED:
+                raise RuntimeError(f"Current state is {salobj.State(ss.summaryState)}.")
+
+            getattr(self, comp).evt_summaryState.set_put(summaryState=salobj.State.ENABLED)
```

Comment:
Use `ss` instead of `getattr(self, comp)` here?

It looks like most of your callbacks that take a `comp` argument use the same pattern and could use the same simplification.